### PR TITLE
Add/Update AWS Security Hub > Foundational Security Best Practices [FSBP v1.0.0] [28-07-2023] closes #703

### LINKED ISSUE #703: Add/Update AWS Security Hub > Foundational Security Best Practices [FSBP v1.0.0] [28-07-2023]
**Is your feature request related to a problem? Please describe.**

[Document history for the AWS Security Hub User Guide](https://docs.aws.amazon.com/securityhub/latest/userguide/doc-history.html)

1. [[Athena.1] Athena workgroups should be encrypted at rest](https://docs.aws.amazon.com/securityhub/latest/userguide/athena-controls.html#athena-1)
2. [[DocumentDB.1] Amazon DocumentDB clusters should be encrypted at rest](https://docs.aws.amazon.com/securityhub/latest/userguide/documentdb-controls.html#documentdb-1)
3. [[DocumentDB.2] Amazon DocumentDB clusters should have an adequate backup retention period](https://docs.aws.amazon.com/securityhub/latest/userguide/documentdb-controls.html#documentdb-2)
4. [[Neptune.1] Neptune DB clusters should be encrypted at rest](https://docs.aws.amazon.com/securityhub/latest/userguide/neptune-controls.html#neptune-1)
5. [[Neptune.2] Neptune DB clusters should publish audit logs to CloudWatch Logs](https://docs.aws.amazon.com/securityhub/latest/userguide/neptune-controls.html#neptune-2)
6. [[Neptune.3] Neptune DB cluster snapshots should not be public](https://docs.aws.amazon.com/securityhub/latest/userguide/neptune-controls.html#neptune-3)
7. [[Neptune.4] Neptune DB clusters should have deletion protection enabled](https://docs.aws.amazon.com/securityhub/latest/userguide/neptune-controls.html#neptune-4)
8. [[Neptune.5] Neptune DB clusters should have automated backups enabled](https://docs.aws.amazon.com/securityhub/latest/userguide/neptune-controls.html#neptune-5)
9. [[Neptune.6] Neptune DB cluster snapshots should be encrypted at rest](https://docs.aws.amazon.com/securityhub/latest/userguide/neptune-controls.html#neptune-6)
10. [[Neptune.7] Neptune DB clusters should have IAM database authentication enabled](https://docs.aws.amazon.com/securityhub/latest/userguide/neptune-controls.html#neptune-7)
11. [[Neptune.8] Neptune DB clusters should be configured to copy tags to snapshots](https://docs.aws.amazon.com/securityhub/latest/userguide/neptune-controls.html#neptune-8)
12. [[RDS.27] RDS DB clusters should be encrypted at rest](https://docs.aws.amazon.com/securityhub/latest/userguide/rds-controls.html#rds-27)

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by khushboo9024**
### Checklist

- [x] [[Athena.1] Athena workgroups should be encrypted at rest](https://docs.aws.amazon.com/securityhub/latest/userguide/athena-controls.html#athena-1)
- [x] [[DocumentDB.1] Amazon DocumentDB clusters should be encrypted at rest](https://docs.aws.amazon.com/securityhub/latest/userguide/documentdb-controls.html#documentdb-1)
- [x] [[DocumentDB.2] Amazon DocumentDB clusters should have an adequate backup retention period](https://docs.aws.amazon.com/securityhub/latest/userguide/documentdb-controls.html#documentdb-2)
- [x] [[Neptune.1] Neptune DB clusters should be encrypted at rest](https://docs.aws.amazon.com/securityhub/latest/userguide/neptune-controls.html#neptune-1)
- [x] [[Neptune.2] Neptune DB clusters should publish audit logs to CloudWatch Logs](https://docs.aws.amazon.com/securityhub/latest/userguide/neptune-controls.html#neptune-2)
- [x] [[Neptune.3] Neptune DB cluster snapshots should not be public](https://docs.aws.amazon.com/securityhub/latest/userguide/neptune-controls.html#neptune-3)
- [x] [[Neptune.4] Neptune DB clusters should have deletion protection enabled](https://docs.aws.amazon.com/securityhub/latest/userguide/neptune-controls.html#neptune-4)
- [x] [[Neptune.5] Neptune DB clusters should have automated backups enabled](https://docs.aws.amazon.com/securityhub/latest/userguide/neptune-controls.html#neptune-5)
- [x] [[Neptune.6] Neptune DB cluster snapshots should be encrypted at rest](https://docs.aws.amazon.com/securityhub/latest/userguide/neptune-controls.html#neptune-6)
- [x] [[Neptune.7] Neptune DB clusters should have IAM database authentication enabled](https://docs.aws.amazon.com/securityhub/latest/userguide/neptune-controls.html#neptune-7)
- [x] [[Neptune.8] Neptune DB clusters should be configured to copy tags to snapshots](https://docs.aws.amazon.com/securityhub/latest/userguide/neptune-controls.html#neptune-8)
- [x] [[RDS.27] RDS DB clusters should be encrypted at rest](https://docs.aws.amazon.com/securityhub/latest/userguide/rds-controls.html#rds-27)

### CODE REVIEW

**Review by rajlearner17 — APPROVED** [review]
LGTM

**Review by madhushreeray30 — APPROVED** [review]
LGTM 👍

File: conformance_pack/docdb.sp [hunk 1]
```diff
@@ -33,4 +33,39 @@ query "docdb_cluster_instance_logging_enabled" {
...
+
+query "docdb_cluster_backup_retention_period_7_days" {
+  sql = <<-EOQ
+    select
+      arn as resource,
+      case
+        when backup_retention_period >= 7 then 'ok'
+        else 'alarm'
+      end as status,
+      title || ' backups retention period is ' || backup_retention_period || ' day(s).' as reason
```
Comment (misraved) [inline conformance_pack/docdb.sp:65]:
```suggestion
      title || ' backup retention period is ' || backup_retention_period || ' day(s).' as reason
```

File: conformance_pack/neptune.sp [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+query "neptune_db_cluster_automated_backup_enabled" {
+  sql = <<-EOQ
+    select
+      arn as resource,
+      case
+        when backup_retention_period >= 7 then 'ok'
+        else 'alarm'
+      end as status,
+      case
+        when backup_retention_period >= 7 then title || ' automated backups enabled.'
```
Comment (misraved) [inline conformance_pack/neptune.sp:112]: In this query we are checking for `backup_retention_period`, is this correct per FSBP documentation?

**Review by misraved — COMMENTED** [review]
Please take a look at the review comment. Thanks!!

File: conformance_pack/neptune.sp (hunk 2 above)
Comment (madhushreeray30) [reply conformance_pack/neptune.sp:112]: yes @misraved the document link is > https://docs.aws.amazon.com/securityhub/latest/userguide/neptune-controls.html#neptune-5

**Review by misraved — APPROVED** [review]

**Pull request merged**